js/src/common/utils/mapRoutes.ts

```diff
@@ -7,7 +7,7 @@
  * @param {String} [basePath]
  * @return {Object}
  */
-export default function mapRoutes(routes, basePath = '') {
+export default function mapRoutes(routes: Object, basePath: string = ''): Object {
```

Comment:
For basePath, doesn't it automatically detect that it's a string based on the default type?

Reply:
Routes here can contain typed properties, see `{ path: string, component: Component }` or something similar.

Reply:
`mithril.Route` as type for routes and `Partial<mithril.Route>` as return type.

Reply:
Does `Mithril.Route` work here? I'm not sure what those typings are exactly 🤔 as we don't follow anything specific to Mithril for `mapRoutes` arguments, I'm unsure here.

Reply:
No ts error is thrown on using the same type.